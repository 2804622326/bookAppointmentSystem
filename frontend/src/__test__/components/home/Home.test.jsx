import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import Home from '../../../components/home/Home.jsx';

jest.mock('../../../components/veterinarian/VeterinarianService.js', () => ({
  getVeterinarians: jest.fn().mockResolvedValue({ data: [{ id: 1, firstName: 'A', lastName: 'B' }] }),
}));

test('renders home and loads vets', async () => {
  const { asFragment } = render(
    <MemoryRouter>
      <Home />
    </MemoryRouter>
  );
  await waitFor(() => screen.getByText(/who we are/i));
  expect(screen.getByText(/our services/i)).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
