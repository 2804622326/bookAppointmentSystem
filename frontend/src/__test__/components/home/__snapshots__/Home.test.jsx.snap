// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home and loads vets 1`] = `
<DocumentFragment>
  <div
    class="home-container mt-5 container"
  >
    <div
      class="row"
    >
      <div
        class="mb-3 col-md-6"
      >
        <div
          class="card"
        >
          <img
            alt="About Us"
            class="card-img-top hero-image"
            src="test-file-stub"
          />
          <div
            class="card-body"
          >
            <h2
              class="text-info"
            >
              Who We Are
            </h2>
            <div
              class="card-title h5"
            >
              Comprehensive Care for your Furry Friends
            </div>
            <p
              class="card-text"
            >
              At Universal Pet Care, we believe every pet deserves the best. Our team of dedicated professionals is here to ensure your pet's health and happiness through comprehensive veterinary services. With decades of combined experience, our veterinarians and support staff are committed to providing personalized care tailored to the unique needs of each pet.our veterinarians and support staff are committed to providing personalized care tailored to the unique needs of each pet.
            </p>
            <p
              class="card-text"
            >
              We offer a wide range of services, from preventive care and routine check-ups to advanced surgical procedures and emergency care. Our state-of-the-art facility is equipped with the latest in veterinary technology, which allows us to deliver high-quality care with precision and compassion. We offer a wide range of services, from preventive care and routine check-ups to advanced surgical procedures and emergency care. Our state-of-the-art facility is equipped with the latest in veterinary technology, which allows us to deliver high-quality care with precision and compassion. Our state-of-the-art facility is equipped with the latest in veterinary technology, which allows us to deliver high-quality care.
            </p>
            <button
              class="btn btn-outline-info"
              type="button"
            >
               Meet Our veterinarians
            </button>
          </div>
        </div>
      </div>
      <div
        class="mb-3 col-md-6"
      >
        <div
          class="service-card card"
        >
          <img
            alt="About Us"
            class="card-img-top hero-image"
            src="test-file-stub"
          />
          <div
            class="card-body"
          >
            <h2
              class="text-info"
            >
              Our Services
            </h2>
            <div
              class="card-title h5"
            >
              What we do
            </div>
            <div
              class="services-list list-group"
            >
              <div
                class="list-group-item"
              >
                Veterinary Check-ups
              </div>
              <div
                class="list-group-item"
              >
                Emergency Surgery
              </div>
              <div
                class="list-group-item"
              >
                Pet Vaccinations
              </div>
              <div
                class="list-group-item"
              >
                Dental Care
              </div>
              <div
                class="list-group-item"
              >
                Spaying and Neutering
              </div>
              <div
                class="list-group-item"
              >
                And many more...
              </div>
            </div>
            <p
              class="mt-3 card-text"
            >
              From routine check-ups to emergency surgery, our full range of veterinary services ensures your pet's health is in good hands.
            </p>
            <button
              class="btn btn-outline-info"
              type="button"
            >
               Meet Our veterinarians
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="card mb-5"
    >
      <h5>
        What people are saying about 
        <span
          class="text-info"
        >
           Universal Pet Care
        </span>
         Veterinarians
      </h5>
      <hr />
      <main>
        <div
          class="carousel slide"
        >
          <div
            class="carousel-indicators"
          >
            <button
              aria-current="true"
              aria-label="Slide 1"
              class="active"
              data-bs-target=""
              type="button"
            />
          </div>
          <div
            class="carousel-inner"
          >
            <div
              class="active carousel-item"
            >
              <div
                class="align-items-center row"
              >
                <div
                  class="text-center col-md-4 col-12"
                >
                  <img
                    alt="photo"
                    class="card-img"
                    src="test-file-stub"
                    style="max-width: 400px; max-height: 400px; object-fit: contain;"
                  />
                </div>
                <div
                  class="col-md-8 col-12"
                >
                  <div>
                    <span
                      class="me-2 ms-2"
                    >
                      <svg
                        color="#ffc107"
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        style="color: rgb(255, 193, 7);"
                        viewBox="0 0 536 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M508.55 171.51L362.18 150.2 296.77 17.81C290.89 5.98 279.42 0 267.95 0c-11.4 0-22.79 5.9-28.69 17.81l-65.43 132.38-146.38 21.29c-26.25 3.8-36.77 36.09-17.74 54.59l105.89 103-25.06 145.48C86.98 495.33 103.57 512 122.15 512c4.93 0 10-1.17 14.87-3.75l130.95-68.68 130.94 68.7c4.86 2.55 9.92 3.71 14.83 3.71 18.6 0 35.22-16.61 31.66-37.4l-25.03-145.49 105.91-102.98c19.04-18.5 8.52-50.8-17.73-54.6zm-121.74 123.2l-18.12 17.62 4.28 24.88 19.52 113.45-102.13-53.59-22.38-11.74.03-317.19 51.03 103.29 11.18 22.63 25.01 3.64 114.23 16.63-82.65 80.38z"
                        />
                      </svg>
                      <svg
                        color="#ffc107"
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        style="color: rgb(255, 193, 7);"
                        viewBox="0 0 576 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
                        />
                      </svg>
                      <svg
                        color="#ffc107"
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        style="color: rgb(255, 193, 7);"
                        viewBox="0 0 576 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
                        />
                      </svg>
                      <svg
                        color="#ffc107"
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        style="color: rgb(255, 193, 7);"
                        viewBox="0 0 576 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
                        />
                      </svg>
                      <svg
                        color="#ffc107"
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        style="color: rgb(255, 193, 7);"
                        viewBox="0 0 576 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
                        />
                      </svg>
                    </span>
                  </div>
                  <div>
                    <p
                      class="text-success"
                    >
                      Dr. A B
                    </p>
                  </div>
                  <p />
                  <p>
                    <span
                      class="text-info"
                    >
                      Dr. A B is a .
                    </span>
                    Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime mollitia, molestiae quas vel sint commodi repudiandae consequuntur voluptatum laborum numquam blanditiis harum quisquam eius sed odit fugiat iusto fuga praesentium optio, eaque rerum! Provident similique accusantium nemo autem. Veritatis obcaecati tenetur iure corporis!
                  </p>
                  <p>
                    <a
                      class="me-3 link-2"
                      href="/vet-reviews/1/veterinarian"
                    >
                      What are people saying about
                    </a>
                    Dr. A B ?
                  </p>
                  <p>
                    <a
                      class="me-3"
                      href="/doctors"
                    >
                      Meet all Veterinarians
                    </a>
                  </p>
                </div>
              </div>
            </div>
          </div>
          <a
            class="carousel-control-prev"
            href="#"
            role="button"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="carousel-control-prev-icon"
            />
            <span
              class="visually-hidden"
            >
              Previous
            </span>
          </a>
          <a
            class="carousel-control-next"
            href="#"
            role="button"
            tabindex="0"
          >
            <span
              aria-hidden="true"
              class="carousel-control-next-icon"
            />
            <span
              class="visually-hidden"
            >
              Next
            </span>
          </a>
        </div>
      </main>
    </div>
  </div>
</DocumentFragment>
`;
